File: 백준/Bronze/2455. 지능형 기차/지능형 기차.cc
#include <iostream>
using namespace std;

int main() {
	
	int left = 0;
	int max = 0;

	for (int i = 0; i < 4; i++)
	{
		int out, in;
		cin >> out >> in;

		left += in - out;

		if (left >= max)
		{
			max = left;
		}
	}

	cout << max << endl;
}

File: 백준/Bronze/2455. 지능형 기차/지능형 기차.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int left = 0;
        int max = 0;

        for (int i = 0; i < 4; i++) {
            int out = sc.nextInt();
            int in = sc.nextInt();

            left += in - out;

            if (left >= max) {
                max = left;
            }
        }

        System.out.println(max);
    }
}